scripts/prebuild/mdx-transforms/add-version-to-internal-links/add-version-to-internal-links.mjs

```diff
@@ -65,8 +73,18 @@ export const rewriteInternalLinksPlugin = ({ entry, versionMetadata }) => {
 		return flatMap(tree, (node) => {
 			// Check if the node is a link and matches the pattern for links to rewrite
 			if (node.type === 'link' && isLinkToRewritePattern.test(node.url)) {
-				// Replace the matched part of the URL with the versioned path
-				node.url = node.url.replace(replacePattern, `/$1/${cleanVersion}$2`)
+				const splitUrl = node.url.split('/')
+				const hasVersionInPath = splitUrl.find((el) => {
+					return (
+						TFE_VERSION_IN_PATH_REGEXP.test(el) ||
+						VERSION_IN_PATH_REGEX.test(el) ||
+						NO_V_VERSION_IN_PATH_REGEX.test(el)
+					)
+				})
+				// Replace the matched part of the URL with the versioned path if no version is present
+				if (!hasVersionInPath) {
+					node.url = node.url.replace(replacePattern, `/$1/${cleanVersion}$2`)
```

Comment:
Here be dragons....

We might want to be careful here with this logic as this has caused issues on dev portal URL slugs that contain versions. You can see an example of this [here](https://developer.hashicorp.com/vault/docs/v1.18.x/upgrading/upgrade-to-1.13.x) where the links in the dropdown get truncated to `https://developer.hashicorp.com/vault/docs/v1.19.x/upgrading/upgrade-to-`.

I have a dev portal PR open [here](https://github.com/hashicorp/dev-portal/pull/2841). I don't know how likely we are to run into that here, but it just seemed worth mentioning.